server/events/output_updater.go

```diff
@@ -0,0 +1,84 @@
+package events
+
+import (
+	"github.com/runatlantis/atlantis/server/core/config/valid"
+	"github.com/runatlantis/atlantis/server/events/command"
+	"github.com/runatlantis/atlantis/server/events/models"
+	"github.com/runatlantis/atlantis/server/events/vcs"
+)
+
+type CommitOutputUpdater interface {
+	Update(ctx *command.Context, cmd PullCommand, res command.Result)
+}
+
+func NewCommitOutputUpdater(client vcs.Client, markdownRenderer *MarkdownRenderer, globalCfg valid.GlobalCfg, useCheckApi bool, hidePrevComments bool) CommitOutputUpdater {
+	if useCheckApi {
+		return &ChecksCommitOutputUpdater{
+			VCSClient:        client,
+			MarkdownRenderer: markdownRenderer,
+			GlobalCfg:        globalCfg,
+		}
+	}
+
+	return &PullCommitOutputUpdater{
+		VCSClient:            client,
+		MarkdownRenderer:     markdownRenderer,
+		GlobalCfg:            globalCfg,
+		HidePrevPlanComments: hidePrevComments,
+	}
+}
+
+type ChecksCommitOutputUpdater struct {
```

Comment:
Checks only applies to github, this is kind of weird that developers are going to be aware of it at this level (outside of the client).

it would be great if we could hide this as an implementation detail of the client itself.